From 6bf665586c0cd4164c9a95b74e70a0ae9d4e8512 Mon Sep 17 00:00:00 2001
From: Sean V Kelley <sean.v.kelley@linux.intel.com>
Date: Mon, 3 Dec 2018 09:25:08 -0800
Subject: [PATCH] Install internal fonts to different location

2019-08-23: Rebased for upstream refactoring in 2.80
2021-02-21: Revised patch subject; rebased for upstream refactoring in
            2.91.2

Signed-off-by: Li, Guangli <guangli.li@intel.com>
Signed-off-by: Patrick McCarty <patrick.mccarty@intel.com>
---
 source/blender/blenfont/intern/blf_font_default.cc | 3 +--
 source/creator/CMakeLists.txt                      | 4 ++--
 2 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/source/blender/blenfont/intern/blf_font_default.cc b/source/blender/blenfont/intern/blf_font_default.cc
index 070818d6131a..e8e0e9a84b6d 100644
--- a/source/blender/blenfont/intern/blf_font_default.cc
+++ b/source/blender/blenfont/intern/blf_font_default.cc
@@ -23,8 +23,7 @@
 
 static int blf_load_font_default(const char *filename, const bool unique)
 {
-  const std::optional<std::string> dir = BKE_appdir_folder_id(BLENDER_DATAFILES,
-                                                              BLF_DATAFILES_FONTS_DIR);
+  const std::optional<std::string> dir = "/usr/share/fonts/blender";
   if (!dir.has_value()) {
     fprintf(stderr,
             "%s: 'fonts' data path not found for '%s', will not be able to display text\n",
diff --git a/source/creator/CMakeLists.txt b/source/creator/CMakeLists.txt
index 23832f21329d..1be9c0605377 100644
--- a/source/creator/CMakeLists.txt
+++ b/source/creator/CMakeLists.txt
@@ -473,8 +473,8 @@ endif()
 
 # fonts
 install(
-  DIRECTORY ${CMAKE_SOURCE_DIR}/release/datafiles/fonts
-  DESTINATION ${TARGETDIR_VER}/datafiles
+  DIRECTORY ${CMAKE_SOURCE_DIR}/release/datafiles/fonts/
+  DESTINATION ${CMAKE_INSTALL_PREFIX}/share/fonts/blender
 )
 
 # localization
-- 
2.44.0

